core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -411,188 +315,140 @@ class SpendReportingServiceSpec
         verifyWorkspaceCategorySubAggregation(workspaceAggregation)
       case categoryAggregation @ SpendReportingAggregation(SpendReportingAggregationKeys.Category, _) =>
         verifyCategoryAggregation(
           categoryAggregation,
-          expectedComputeCost = SpendReportingTestData.SubAggregation.computeTotalCostRounded,
-          expectedStorageCost = SpendReportingTestData.SubAggregation.storageTotalCostRounded,
-          expectedOtherCost = SpendReportingTestData.SubAggregation.otherTotalCostRounded
+          expectedCompute = TestData.SubAggregation.computeTotalCostRounded,
+          expectedStorage = TestData.SubAggregation.storageTotalCostRounded,
+          expectedOther = TestData.SubAggregation.otherTotalCostRounded
         )
       case _ => fail("unexpected aggregation key found")
     }
   }
 
-  private def verifyCategoryAggregation(categoryAggregation: SpendReportingAggregation,
-                                        expectedComputeCost: BigDecimal,
-                                        expectedStorageCost: BigDecimal,
-                                        expectedOtherCost: BigDecimal
-  ) =
-    categoryAggregation.spendData.map { spendDataForCategory =>
+  def verifyCategoryAggregation(
+    aggregation: SpendReportingAggregation,
+    expectedCompute: String,
+    expectedStorage: String,
+    expectedOther: String
+  ): Unit =
+    aggregation.spendData.foreach { spendDataForCategory =>
       val category = spendDataForCategory.category.getOrElse(fail("results not parsed correctly"))
 
       withClue(s"total $category cost was incorrect") {
         if (category.equals(TerraSpendCategories.Compute)) {
-          spendDataForCategory.cost shouldBe expectedComputeCost.toString
+          spendDataForCategory.cost shouldBe expectedCompute
         } else if (category.equals(TerraSpendCategories.Storage)) {
-          spendDataForCategory.cost shouldBe expectedStorageCost.toString
+          spendDataForCategory.cost shouldBe expectedStorage
         } else if (category.equals(TerraSpendCategories.Other)) {
-          spendDataForCategory.cost shouldBe expectedOtherCost.toString
+          spendDataForCategory.cost shouldBe expectedOther
         } else {
           fail(s"unexpected category found in spend results - $spendDataForCategory")
         }
       }
     }
 
-  private def verifyWorkspaceCategorySubAggregation(topLevelAggregation: SpendReportingAggregation) = {
+  def verifyWorkspaceCategorySubAggregation(topLevelAggregation: SpendReportingAggregation): Unit = {
     topLevelAggregation.aggregationKey shouldBe SpendReportingAggregationKeys.Workspace
 
-    topLevelAggregation.spendData.map { spendData =>
-      val workspaceGoogleProject = spendData.googleProjectId.getOrElse(fail("spend results not parsed correctly")).value
-      val subAggregation = spendData.subAggregation.getOrElse(fail("spend results not parsed correctly"))
+    topLevelAggregation.spendData.foreach { spendData =>
```

Comment:
For my own edification, why is foreach better than map here?

Reply:
Is it just that we don't actually want to return anything, and map returns a sequential?

Reply:
map is for when you want to do something with the result. 
foreach is to execute for side effects

In this case, what we get back from the execution is just a list of type Unit placeholder type, which is something along the lines of void in other languages. 
So there isn't actually anything we can do with the result anyway - this just makes the result agree how we're using it.